hvat/src/hyperview/asset_api_functions.rs

```diff
@@ -407,221 +422,81 @@ pub async fn search_assets_async(
-                    _ => None,
-                }
-            })
-            .collect();
-
-        kv.iter().for_each(|(k, v)| {
-            let subquery = json!({
-              "nested": {
-                "path": "stringCustomProperties",
-                "inner_hits": {},
-                "query": {
-                  "bool": {
-                    "must": [
-                      {
-                        "match": {
-                          "stringCustomProperties.name": { "query": k }
-                        }
-                      },
-                      {
-                        "match": {
-                          "stringCustomProperties.value": {
-                            "query": v,
-                            "lenient": true
-                          }
-                        }
-                      }
-                    ]
-                  }
-                }
-              }
-            });
-
-            search_query["query"]["bool"]["must"]
-                .as_array_mut()
-                .unwrap()
-                .push(subquery);
-        });
+    if let Some(custom_properties) = options.custom_properties {
+        for custom_property in custom_properties {
+            if let Some((custom_property_key_name, custom_property_key_value)) = custom_property.split_once('=') {
+                 let custom_property_key_attribute = format!("customProperty_{} ", custom_property_key_name.trim());
+                filters.push(format!("{} = '{}'", custom_property_key_attribute, custom_property_key_value.trim()));
+            } else {
+                eprintln!("Custom asset property filter was formatted incorrectly. Skipping... '{}'", custom_property);
+            }
+        }
     }
 
     if let Some(id_guid) = options.id {
-        let subquery = json!({ "match": { "id": { "query": id_guid, "lenient": true } } });
-        search_query["query"]["bool"]["must"]
-            .as_array_mut()
-            .unwrap()
-            .push(subquery);
+        let id_query = format!("id = '{}'", id_guid);
+        filters.push(id_query);
     }
 
     if let Some(manufacturer) = options.manufacturer {
-        let subquery = json!({ "match": { "manufacturerNameSearchableProperty": { "query": manufacturer, "lenient": true } } });
-        search_query["query"]["bool"]["must"]
-            .as_array_mut()
-            .unwrap()
-            .push(subquery);
+        let manufacturer_name_query = format!("manufacturerName = '{}'", manufacturer);
+        filters.push(manufacturer_name_query);
     }
 
     if let Some(product) = options.product {
-        let subquery = json!({ "match": { "productNameSearchableProperty": { "query": product, "lenient": true } } });
-        search_query["query"]["bool"]["must"]
-            .as_array_mut()
-            .unwrap()
-            .push(subquery);
+        let product_name_query = format!("productName CONTAINS '{}'", product);
```

Comment:
Using `CONTAINS` here specifically to be a litte more user friendly.

If you provide "poweredge" as a name, all models starting with "PowerEdge" will be netted--which seemed like a valid use case to support.

If you provide  "PowerEdge R420", the match will be more strict.

Reply:
Good suggestion, thank you!